card/lib/card/view_name.rb

```diff
@@ -0,0 +1,44 @@
+# -*- encoding : utf-8 -*-
+require_dependency 'card/env'
+
+require 'smart_name'
+
+class Card
+  class ViewName < SmartName
+    @@name2viewnameobject = {}
+
+    class << self
+      def new obj
+        return obj if self.class===obj
+        str = Array===obj ? obj*joint : obj.to_s
+        if known_name = @@name2viewnameobject[str]
+          known_name
+        else
+          super str.strip
+        end
+      end
+    end
+
+    def initialize str
+      @s = str.to_s.strip
+      @s = @s.encode('UTF-8') if RUBYENCODING
+      @key = if @s.index(self.class.joint)
+          @parts = @s.split(/\s*#{JOINT_RE}\s*/)
+          @parts << '' if @s[-1,1] == self.class.joint
+          @simple = false
+          @parts.map { |p| p.to_name.key } * self.class.joint
+        else
+          @parts = [str]
+          @simple = true
+          str.empty? ? '' : simple_key
+        end
+      @@name2viewnameobject[str] = self
+    end
+
+    def simple_key
+      decoded.underscore.gsub(/[^#{OK4KEY_RE}]+/,'_').split(/_+/).reject(&:empty?)*'_'
+    end
```

Comment:
So, you will move the gem to the wagn repo so we'll have four now?  It shouldn't be that big of a deal to maintain either way, but I think together makes sense and follows that rails platform pattern.  Folks could still use any subset of the 4 like they would will rails gems.
I think there could be some real advantages to using name logic for views.  It's much stronger if we actually have cards for views, but I think it is mostly a good thing.  We are conflating ruby (and rails) inflections for constants vs. identifiers where singular/plural in particular is used differently with our name processing which is about 1) folding equivalence classes (as defined by key identity) and 2) translating between presentation names and internal (code) names.
Oh, that reminds me, codenames really are arbitrary.  Generally we start with the codename being the key for the default name, but there is no logical requirement for this.  We do the singular processing for 1), to fold from all the possible names to the much smaller space of keys, but I think this issue is all about 2), having a stable codename.